### ISSUE TITLE
torch.cat on empty tensor is bogus

### ISSUE THREAD

**Issue by zou3519** [pytorch/pytorch#86918]
### 🐛 Describe the bug

torch.cat on Tensors of shape (0,) ignores the dim argument to cat:
```
x = torch.randn([0])
result = torch.cat([x, x], dim=999)
```

It should most likely error in this case.

### Versions

master

**Comment by SsnL** [comment]
but it should still work if `dim=0`, right?

**Comment by zou3519** [comment]
Right, it should work on dim=0 but error out on the rest

**Comment by ngimel** [comment]
`cat` has a legacy exception where the tensor of shape (0,) can be catted with anything, I guess when only those "exception" tensors are present no further checks are run.
